### PR TITLE
initial implementation

### PR DESCRIPTION
**Pull request by mcalhoun**
## what

The initial implementation of the `cloudposse/utils` provider. The first functionality is a `data_source` that takes a list of maps as input and outputs a single, deep-merged map.

## why

We have a need to deep merge multiple configuration files from our reference architecture "stacks"

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,397 @@
...
+
+```hcl
+terraform {
+  required_providers {
+    utils = {
+      source = "cloudposse/utils"
+      version = "0.0.1-alpha5"
+    }
+  }
+}
```
Comment (osterman) [inline README.md:74]: Let's have an in-line example as well.

Comment (mcalhoun) [reply README.md:74]: 6f34114642c12dc7c1302b91d7b635bd94ff67ec

**Review by osterman — APPROVED** [review]

File: main.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+	"flag"
+	"log"
+
+	"github.com/cloudposse/terraform-provider-utils/internal/provider"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/plugin"
+	
+)
+
+var (
+	version string = "dev"
```
Comment (aknysh) [inline main.go:14]: what's the `"dev" version?
Should we change it to prod or something different?

**Review by aknysh — COMMENTED** [review]
please see comments

File: main.go (hunk 2 above)
Comment (mcalhoun) [reply main.go:14]: That's just used locally in tests.

**Pull request merged**